subprojects/core-api/src/main/java/org/gradle/api/credentials/HttpHeaderCredentials.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.credentials;
+
+import javax.annotation.Nullable;
+
+/**
+ * Header credentials that can be used to login to something protected by using HTTP header.
+ */
+public interface HttpHeaderCredentials extends Credentials {
+    @Nullable
+    String getHeader();
+    void setHeader(@Nullable String header);
```

Comment:
I think we might as well support multiple headers, so the setter should take and the getter should return a `Map<String, String>`.

Reply:
I see, thanks for the explanation!